lib/srv/db/common/session.go

```diff
@@ -94,3 +95,27 @@ func (c *Session) WithUserAndDatabase(user string, defaultDatabase string) *Sess
 	copy.DatabaseName = defaultDatabase
 	return copy
 }
+
+// CheckUsernameForAutoUserProvisioning checks the username when using
+// auto-provisioning.
+//
+// When using auto-provisioning, force the database username to be same
+// as Teleport username. If it's not provided explicitly, some database
+// clients get confused and display incorrect username.
+func (c *Session) CheckUsernameForAutoUserProvisioning() error {
+	if !c.AutoCreateUserMode.IsEnabled() {
+		return nil
+	}
+
+	if c.DatabaseUser == c.Identity.Username {
+		return nil
+	}
+
+	if c.AuthContext != nil && authz.IsRemoteUser(*c.AuthContext) {
+		return trace.AccessDenied("please use your mapped remote username (%q) to connect instead of %q",
+			c.Identity.Username, c.DatabaseUser)
```

Comment:
Nit: 
Could we add a short docs troubleshooting update and explain what this error means. 

User needs to be aware of remote clusters setup and know that username on remote cluster is mapped to `remote-main-cluster-username` without this this massage is quite vague. 


We can just add a troubleshooting error "please use your mapped remote username" and explain this error.

Reply:
we added a note on the long name here: https://github.com/gravitational/teleport/pull/36648/files#diff-026e51d24f3932c0b0dca15ed5e7c96362d9217a96311bc3c69cff193f4e2356

but a troubleshooting will definitely help when that error pops up. will do separately. thanks!